main/lsp/lsp.h

```diff
@@ -52,41 +53,9 @@ struct LSPResult {
                           bool canceled = false);
 };
 
-/**
- * A version of ShowOperation that works on the preprocessor thread.
- */
-class ShowOperationPreprocessorThread final {
```

Comment:
No need for a specialized class anymore. The preprocessor thread can use the regular ShowOperation class now; it just provides a custom LSPOutput implementation.